fcrepo/fcrepo
# FCREPO-3830: Update ACL for Transaction Endpoint

**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3830

# What does this Pull Request do?

Updates the ACL logic to allow for checks against the fcr:tx endpoint

# Changes

* Iterate default acls by subject to create multiple WebACAuthorizations
* Allow getting of roles for non-resource endpoints
* Check for match on fcr:tx when getting roles
* Update WebACFilter logic for transactions

# How should this be tested?

* Verify build works and all tests pass
  * mvn clean verify
* Run Fedora in your preferred way
* Test that with the default ACL, only fedoraAdmins should be able to interact with the fcr:tx endpoint
* Stop Fedora and update the default root ACL to include fcr:tx permissions, e.g.
```
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix acl: <http://www.w3.org/ns/auth/acl#> .
@prefix foaf: <http://xmlns.com/foaf/0.1/> .
@prefix fedora: <http://fedora.info/definitions/v4/repository#> .
@prefix webac: <http://fedora.info/definitions/v4/webac#> .

<info:fedora/fcr:acl> a webac:Acl .

<info:fedora/fcr:acl#authz> a acl:Authorization ;
   rdfs:label "Root Authorization" ;
   rdfs:comment "By default, all non-Admin agents (foaf:Agent) only have read access (acl:Read) to the repository" ;
   acl:agentClass foaf:Agent ;
   acl:mode acl:Read ;
   acl:accessTo <info:fedora> ;
   acl:default <info:fedora> .

<info:fedora/fcr:tx#authz_read_write> a acl:Authorization ;
   rdfs:label "Test Tx Authorization" ;
   rdfs:comment "Provide read write access to the transaction endpoint" ;
   acl:agentClass acl:AuthenticatedAgent ;
   acl:mode acl:Read, acl:Write ;
   acl:accessTo <info:fedora/fcr:tx> ;
   acl:default <info:fedora/fcr:tx> .
``` 
* Run Fedora again and verify that authenticated users can now interact with the fcr:tx endpoint

# Notes

I left the default root acl the same. I've been thinking about whether or not we want to update it but haven't come to a decision on my own. Either way we'll want to update the documentation to showcase this scenario.

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers
